ferrocene/tools/self-test/src/targets.rs

```diff
@@ -109,11 +111,55 @@ fn check_target(
         false => ["core", "alloc"].as_slice(),
     };
     check_libraries(target, &target_dir, expected_libs)?;
+    check_default_link_args(sysroot, target)?;
```

Comment:
Removing that line does not lead to any failing tests. Maybe we can add a test that verifies it fails when it should, in this or a future PR.